pkg/reconciler/instances/ory/hydra/hydra_test.go

```diff
@@ -0,0 +1,187 @@
+		logger.Debugf("Deploy found ", deploymentOld.Name)
+		err := NewDefaultHydraSyncer().TriggerSynchronization(context.TODO(), client, logger, testNamespase)
+
+		// then
+		require.NoError(t, err)
+		deployment, err2 := client.AppsV1().Deployments(testNamespase).Get(context.TODO(), "ory-hydra-maester", metav1.GetOptions{})
+		require.NoError(t, err2)
+		spec := deployment.Spec
+		restartedAt := spec.Template.Annotations["kubectl.kubernetes.io/restartedAt"]
+		// deployment should not have been restarted
+		require.Equal(t, restartedAt, hydraMasesterPodStartTime.String())
+
+	})
+}
+func Test_GetEarliestStartTime(t *testing.T) {
+	logger := zaptest.NewLogger(t).Sugar()
+	t.Parallel()
+	t.Run("Should find earliest pod starttime out of two", func(t *testing.T) {
+		// given
+		startTimePod1 := time.Date(2021, 10, 10, 10, 10, 10, 10, time.UTC)
+		startTimePod2 := time.Date(2021, 10, 10, 10, 10, 7, 10, time.UTC)
+		kubeclient := fakeClient()
+		addPod(kubeclient, "hydra1", "hydra", startTimePod1, t, v1.PodRunning)
+		addPod(kubeclient, "hydra2", "hydra", startTimePod2, t, v1.PodRunning)
+		client, _ := kubeclient.Clientset()
+
+		// when
+		earliestStartTime, err := getEarliestPodStartTime(context.TODO(), hydraPodName, client, logger, testNamespase)
+
+		// then
+		require.NoError(t, err)
+		require.Equal(t, startTimePod2, earliestStartTime)
+	})
+	t.Run("Should determine earliest pod starttime if both started at the same time", func(t *testing.T) {
+		// given
+		startTime1 := time.Date(2021, 10, 10, 10, 10, 10, 10, time.UTC)
+		startTime2 := time.Date(1900, 10, 10, 10, 10, 10, 10, time.UTC)
+		kubeclient := fakeClient()
+		addPod(kubeclient, "hydra1", "hydra", startTime1, t, v1.PodRunning)
+		addPod(kubeclient, "hydra2", "hydra", startTime1, t, v1.PodRunning)
+		addPod(kubeclient, "hydra3", "hydra", startTime2, t, v1.PodPending)
+		client, _ := kubeclient.Clientset()
+
+		// when
+		earliestStartTime, err := getEarliestPodStartTime(context.TODO(), hydraPodName, client, logger, testNamespase)
+
+		// then
+		require.NoError(t, err)
+		require.Equal(t, earliestStartTime, startTime1)
+	})
+
+	t.Run("Should return error if no running pods found", func(t *testing.T) {
+		// given
+		startTime1 := time.Date(2021, 10, 10, 10, 10, 10, 10, time.UTC)
+		startTime2 := time.Date(1900, 10, 10, 10, 10, 10, 10, time.UTC)
+		kubeclient := fakeClient()
+		addPod(kubeclient, "hydra1", "hydra", startTime1, t, v1.PodFailed)
+		addPod(kubeclient, "hydra2", "hydra", startTime1, t, v1.PodPending)
+		addPod(kubeclient, "hydra3", "hydra", startTime2, t, v1.PodPending)
+		client, _ := kubeclient.Clientset()
+
+		// when
+		_, err := getEarliestPodStartTime(context.TODO(), hydraPodName, client, logger, testNamespase)
+
+		// then
+		require.Error(t, err, "Could not find any running pod for label %s in namespace %s", hydraPodName, testNamespase)
+	})
+	t.Run("Should return error if no pods found", func(t *testing.T) {
+		// given
+		client, _ := fakeClient().Clientset()
```

Comment:
Instead of omitting error you can add one additional check such as: `require.NoError()` - it is more elegant